src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        List<T> machines = new ArrayList<>();
+
+        if (Truck.class.isAssignableFrom(type)) {
+            machines.addAll((List<T>) new TruckProducer().get());
+        } else if (Bulldozer.class.isAssignableFrom(type)) {
+            machines.addAll((List<T>) new BulldozerProducer().get());
+        } else if (Excavator.class.isAssignableFrom(type)) {
+            machines.addAll((List<T>) new ExcavatorProducer().get());
+        }
+
+        return machines;
+    }
+
+    @Override
+    public void fill(List<Object> machines, Machine value) {
+        machines.replaceAll(ignored -> value);
```

Comment:
The size of a list passed to `fill()` method should remain the same. Just replace the old values with the new ones. This implementation is correct, but please ensure the generic type is used for the method parameter.